### PR TITLE
dashboard/feature/add_cuisine

### PR DESCRIPTION
**Pull request by mujtaba912**
implement add cuisines dialog in restaurant screen in client dashboard which have:
- get all cuisines 
- add a cuisine
- delete a cuisine

https://github.com/TheChance101/beep-beep/assets/123612093/961a2d9e-be26-40ca-aef2-65c6d0d069ef

### CODE REVIEW

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantInteractionListener.kt [hunk 1]
```diff
@@ -38,4 +38,15 @@ interface RestaurantInteractionListener : BaseInteractionListener {
...
+    fun onClickAddCuisine()
+
+    fun onClickDeleteCuisine(cuisineName: String)
+
+    fun onAddCuisineDialogCloseRequest()
+
+    fun onAddCuisineDialogAddClicked()
+
+    fun onCuisineNameChange(cuisineName: String)
+
```
Comment (devsadeq) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantInteractionListener.kt:51]: Make your functions names more consistence plz.

**Review by devsadeq — APPROVED** [review]
LGTM

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantInteractionListener.kt (hunk 1 above)
Comment (mujtaba912) [reply client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantInteractionListener.kt]: done

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt [hunk 2]
```diff
@@ -281,4 +313,41 @@ class RestaurantScreenModel(
             )
         }
     }
+
+    override fun onAddCuisineDialogDeleteClicked(cuisineName: String) {
```
Comment (Karrar-Mohammed) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:317]: fix this name

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt [hunk 3]
```diff
@@ -281,4 +313,41 @@ class RestaurantScreenModel(
...
+        tryToExecute(
+            { mangeCuisines.deleteCuisine(cuisineName) },
+            ::onGetCuisinesSuccessfully,
+            ::onError
+        )
+    }
+
+    override fun onAddCuisineDialogCloseRequest() {
```
Comment (Karrar-Mohammed) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:325]: fix this name

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt [hunk 4]
```diff
@@ -281,4 +313,41 @@ class RestaurantScreenModel(
...
+            it.copy(
+                restaurantAddCuisineDialogUiState = it.restaurantAddCuisineDialogUiState.copy(
+                    isVisible = false,
+                    cuisineName = ""
+                )
+            )
+        }
+    }
+
+    override fun onAddCuisineDialogAddClicked() {
```
Comment (Karrar-Mohammed) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:336]: fix this name

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt [hunk 5]
```diff
@@ -281,4 +313,41 @@ class RestaurantScreenModel(
...
+        tryToExecute(
+            { mangeCuisines.addCuisine(state.value.restaurantAddCuisineDialogUiState.cuisineName) },
+            ::onGetCuisinesSuccessfully,
+            ::onError
+        )
+    }
+
+    override fun onAddCuisineDialogCuisineNameChange(cuisineName: String) {
```
Comment (Karrar-Mohammed) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantScreenModel.kt:344]: fix this name

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantUiState.kt [hunk 6]
```diff
@@ -53,3 +54,10 @@ data class RestaurantFilterDropdownMenuUiState(
     val filterPriceLevel: Int = 1,
     val isFiltered: Boolean = false,
 )
+
+data class RestaurantAddCuisineDialogUiState(
+    val isVisible: Boolean = false,
+    val cuisineName: String = "",
+    val cuisines: List<String> = emptyList(),
+    val cuisinesSize: Int = 0
```
Comment (Karrar-Mohammed) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantUiState.kt:62]: we don't need this parameter

Comment (Karrar-Mohammed) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/restaurant/RestaurantUiState.kt:62]: we don't need this parameter we should delete it

**Review by Karrar-Mohammed — CHANGES_REQUESTED** [review]

**Review by Karrar-Mohammed — APPROVED** [review]

**Pull request merged**